backend/src/bork-protocol/utils/token-monitor/token-search.ts

```diff
@@ -0,0 +1,80 @@
+import type { TweetQueueService } from '@/services/twitter/analysis-queue.service';
+import type { TwitterConfigService } from '@/services/twitter/twitter-config-service';
+import type { TwitterService } from '@/services/twitter/twitter-service';
+import type { InterestingToken } from '@/types/token-monitor/token';
+import { elizaLogger } from '@elizaos/core';
+import { SearchMode } from 'agent-twitter-client';
+
+/**
+ * Searches for tweets related to a specific token
+ * @param token The interesting token to search for
+ * @param twitterService Twitter service for searching
+ * @param twitterConfigService Service to get Twitter configuration
+ * @param tweetQueueService Service to queue tweets for processing
+ * @param recentlySearchedTokens Set of tokens that were recently searched
+ * @returns The updated set of recently searched tokens
+ */
+export async function searchTokenTweets(
```

Comment:
name of the function is a bit confusing here because it searches for tweets (only if not previously searched) but then it returns a set?